src/USER-PHONON/dynamical_matrix.cpp

```diff
@@ -0,0 +1,482 @@
+        if (force->dihedral) force->dihedral->compute(eflag,vflag);
+        if (force->improper) force->improper->compute(eflag,vflag);
+    }
+
+    if (force->kspace) {
+        force->kspace->setup();
+        if (kspace_compute_flag) force->kspace->compute(eflag,vflag);
+        else force->kspace->compute_dummy(eflag,vflag);
+    }
+
+    //modify->setup_pre_reverse(eflag,vflag);
+    if (force->newton) comm->reverse_comm();
+
+    openfile("dynmat.dyn");
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DynamicalMatrix::command(int narg, char **arg)
+{
+    MPI_Comm_rank(world,&me);
+
+    if (domain->box_exist == 0)
+        error->all(FLERR,"Dynamical_matrix command before simulation box is defined");
+    if (narg < 2) error->all(FLERR,"Illegal dynamical_matrix command");
+
+    lmp->init();
+
+    // orthogonal vs triclinic simulation box
+
+    triclinic = domain->triclinic;
+
+    if (force->pair && force->pair->compute_flag) pair_compute_flag = 1;
+    else pair_compute_flag = 0;
+    if (force->kspace && force->kspace->compute_flag) kspace_compute_flag = 1;
+    else kspace_compute_flag = 0;
+
+    // group and style
+
+    igroup = group->find(arg[0]);
+    if (igroup == -1) error->all(FLERR,"Could not find dynamical matrix group ID");
+    groupbit = group->bitmask[igroup];
+    dynlen = (group->count(igroup))*3;
+    memory->create(dynmat,int(dynlen),int(dynlen),"dynamic_matrix:dynmat");
+    update->setupflag = 1;
+
+    int style = -1;
+    if (strcmp(arg[1],"regular") == 0) style = REGULAR;
+    else if (strcmp(arg[1],"eskm") == 0) style = ESKM;
+    else error->all(FLERR,"Illegal Dynamical Matrix command");
+
+    // set option defaults
+
+    binaryflag = 0;
+    scaleflag = 0;
+    compressed = 0;
+    file_flag = 0;
+    file_opened = 0;
+    conversion = 1;
+
+    // read options from end of input line
+    if (style == REGULAR) options(narg-3,&arg[3]);  //COME BACK
+    else if (style == ESKM) options(narg-3,&arg[3]); //COME BACK
+    else if (comm->me == 0 && screen) fprintf(screen,"Illegal Dynamical Matrix command\n");
+
+    // move atoms by 3-vector or specified variable(s)
+
+    if (style == REGULAR) {
+        setup();
+        calculateMatrix(arg[2]);
+        if (me ==0) writeMatrix();
+    }
+
+    if (style == ESKM) {
+        setup();
+        convert_units(update->unit_style);
+        conversion = conv_energy/conv_distance/conv_mass;
+        calculateMatrix(arg[2]);
+        if (me ==0) writeMatrix();
+    }
+
+    Finish finish(lmp);
+    finish.end(1);
+}
+
+/* ----------------------------------------------------------------------
+   parse optional parameters
+------------------------------------------------------------------------- */
+
+void DynamicalMatrix::options(int narg, char **arg)
+{
+    if (narg < 0) error->all(FLERR,"Illegal dynamical_matrix command");
+    int iarg = 0;
+    const char* filename;
```

Comment:
just initialize filename here to a default filename and always call openfile below. That will address my concern from above.